File: app/src/main/java/com/example/feel_journal/data/remote/dto/NotificationDtos.kt
package com.example.feel_journal.data.remote.dto

import com.google.gson.annotations.SerializedName

data class CreateNotificationRequest(
    @SerializedName("message") val message: String,
    @SerializedName("date") val date: String
)

data class UpdateNotificationRequest(
    @SerializedName("message") val message: String?,
    @SerializedName("date") val date: String?,
    @SerializedName("isRead") val isRead: Boolean?
)

data class NotificationDto(
    @SerializedName("id") val id: Int,
    @SerializedName("message") val message: String,
    @SerializedName("date") val date: String,
    @SerializedName("userId") val userId: Int,
    @SerializedName("isRead") val isRead: Boolean,
    @SerializedName("createdAt") val createdAt: String
)

File: server/DTOs/NotificationDtos.cs
using System.ComponentModel.DataAnnotations;

namespace FeelJournal.DTOs;

public class NotificationDto
{
    public int Id { get; set; }
    public string Message { get; set; } = string.Empty;
    public string Date { get; set; } = string.Empty;
    public int UserId { get; set; }
    public bool IsRead { get; set; }
    public string CreatedAt { get; set; } = string.Empty;
}

public class CreateNotificationRequest
{
    [Required, StringLength(500)]
    public string Message { get; set; } = string.Empty;

    [Required]
    public string Date { get; set; } = string.Empty;
}

public class UpdateNotificationRequest
{
    [StringLength(500)]
    public string? Message { get; set; }
    public string? Date { get; set; }
    public bool? IsRead { get; set; }
}
